plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/CoreIntrinsics.java

```diff
@@ -187,7 +186,7 @@ private static void registerEmptyNativeInitMethods(final CompilationContext ctxt
     private static StaticIntrinsic setVolatile(CompilationContext ctxt, FieldElement field) {
         Literal voidLiteral = ctxt.getLiteralFactory().zeroInitializerLiteralOfType(ctxt.getTypeSystem().getVoidType());
         return (builder, target, arguments) -> {
-            builder.store(builder.staticField(field), arguments.get(0), MemoryAtomicityMode.VOLATILE);
+            builder.store(builder.staticField(field), arguments.get(0), GlobalRelease);
```

Comment:
According to the (somewhat out-of-date) [JSR-133 Cookbook for Compiler Writers ](http://gee.cs.oswego.edu/dl/jmm/cookbook.html), there needs to be a `StoreLoad` barrier between every volatile store and volatile load.  The only write mode that would include that directly is `GlobalSeqCst`.